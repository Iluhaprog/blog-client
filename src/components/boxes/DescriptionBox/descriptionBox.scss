@import '../../../vars.scss';

.description-box {
    max-width: 600px;
    margin: 40px;
    padding: 20px;
    background-color: $description-bg;
    border-radius: 10px;
    box-shadow: $shadow-2;
    font-family: $main-font;
    border: $border-2;


    ul {
        list-style: none;
        padding: 0;
    }

    &__title {
        font-size: 36px;
        font-weight: 300;
        color: $text-color-8;
    }

    &__description {
        max-height: 150px;
        overflow-y: scroll;
        font-size: 18px;
        word-spacing: 1px;
        line-height: 26px;
        color: $text-color-6;
        padding: 10px;
        border-radius: 5px;
        border: $border-2;

        @media screen and (max-width: 414px) {
            max-height: 300px;
        }
    }
}