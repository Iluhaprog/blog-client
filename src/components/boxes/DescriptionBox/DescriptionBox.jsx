import React from 'react';
import PropTypes from 'prop-types';
import { Column } from '../../containers';
import './descriptionBox.scss';

function DescriptionBox(props) {
    return (
        <div className='description-box'>
            <Column>
                <h1 className='description-box__title'>
                    { props.title }
                </h1>
                <p className='description-box__description'>
                    { props.description }
                </p>
                { props.children }
            </Column>
        </div>
    );
}

DescriptionBox.defaultProps = {
    description: '',
    title: '',
};

DescriptionBox.propTypes = {
    title: PropTypes.string,
    description: PropTypes.string,
    children: PropTypes.node,
};

export default DescriptionBox;
