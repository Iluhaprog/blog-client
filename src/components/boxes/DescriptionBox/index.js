import DescriptionBox from './DescriptionBox';

export {
    DescriptionBox
};