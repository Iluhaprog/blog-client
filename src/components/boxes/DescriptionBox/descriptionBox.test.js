import React from 'react'; 
import { render } from '@testing-library/react';
import DescriptionBox from './DescriptionBox';

test('Should render a description box', () => {
    const { asFragment } = render(
        <DescriptionBox
            title='test title'
            description='test description'
        >
            <div><p>test children</p></div>
        </DescriptionBox>
    );
    expect(asFragment()).toMatchSnapshot();
});