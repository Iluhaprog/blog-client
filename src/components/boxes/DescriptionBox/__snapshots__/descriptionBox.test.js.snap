// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a description box 1`] = `
<DocumentFragment>
  <div
    class="description-box"
  >
    <div
      class="column   "
    >
      <h1
        class="description-box__title"
      >
        test title
      </h1>
      <p
        class="description-box__description"
      >
        test description
      </p>
      <div>
        <p>
          test children
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
